pkg/server/plugin/nodeattestor/aws/iid.go

```diff
@@ -54,30 +57,59 @@ C1haGgSI/A1uZUKs/Zfnph0oEI0/hu1IIJ/SKBDtN5lvmZ/IzbOPIJWirlsllQIQ
 7zvWbGd9c9+Rm3p04oTvhup99la7kZqevJK0QRdD/6NpCKsqP/0=
 -----END CERTIFICATE-----`
 
+// IIDAttestorConfig is the config for IIDAttestorPlugin.
 type IIDAttestorConfig struct {
-	AccessKeyID     string `hcl:"access_key_id"`
-	SecretAccessKey string `hcl:"secret_access_key"`
-	SkipBlockDevice bool   `hcl:"skip_block_device"`
+	AccessKeyID        string `hcl:"access_key_id"`
+	SecretAccessKey    string `hcl:"secret_access_key"`
+	SkipBlockDevice    bool   `hcl:"skip_block_device"`
+	SkipEC2Calling     bool   `hcl:"skip_ec2_calling"`
+	AgentPathTemplate  string `hcl:"agent_path_template"`
+	pathTemplate       *template.Template
+	trustDomain        string
+	awsCaCertPublicKey *rsa.PublicKey
 }
 
+// IIDAttestorPlugin implements node attestation for agents running in aws.
 type IIDAttestorPlugin struct {
-	trustDomain string
+	config *IIDAttestorConfig
+	mtx    sync.RWMutex
```

Comment:
regular `sync.Mutex` seems appropriate since we only use the Write lock (alternatively, change `getConfig` to take the Read lock)